devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,346 @@
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break		
+		case 5:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	result << createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	log.info "zwaveEvent ***** Unhandled Command called, cmd '${cmd}', endpoint '${endpoint}' *****"
+	[descriptionText: "Unhandled $device.displayName: $cmd", isStateChange: true]
+}
+
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping(endpoint = null) {
+	log.info "ping called"
+	refresh(endpoint)
+}
+
+def refresh(endpoint = null) {
+	log.info "refresh called: endpint '${endpoint}' "
+	if(endpoint) {
+		secureEncap(zwave.basicV1.basicGet(), endpoint)
+	} else {
+		def commands = []
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 5)
+		sendCommands(commands,1000)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	log.info "zwaveEvent ConfigurationReport called: ${cmd}"
+	def result = []
+	switch (cmd.parameterNumber) {
+		case 1:
+			state.tempOffset = cmd.scaledConfigurationValue
+			break
+		case 2:
+			state.humidityOffset = cmd.scaledConfigurationValue
+			break
+	}
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, endpoint = null) {
+	log.info "zwaveEvent MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		def type = getNumberOfChildFromModel()
+		log.debug "MultiChannelEndPointReport: type '${type}'"
+		if (type) {
+			addChildSwitches(type)
+		} else {
+			log.debug "child endpoint=$cmd.endPoints"
+			addChildSwitches(cmd.endPoints)
+		}
+	}
+}
+
+def childOn(deviceNetworkId) {
+	log.info "childOn called: deviceNetworkId '${deviceNetworkId}'"
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(0xFF, switchId)
+}
+
+def childOff(deviceNetworkId) {
+	log.info "childOff called: deviceNetworkId '${deviceNetworkId}', value '${value}'"
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(0x00, switchId)
+}
+
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
+
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+
+	def CmdEncap = []
```

Comment:
nit: variables should start with lowercase letters

Reply:
This variables will update from "CmdEncap" to "cmdEncap".